File: sdk/python/feast/registry_store.py
from abc import ABC, abstractmethod

from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto


class RegistryStore(ABC):
    """
    RegistryStore: abstract base class implemented by specific backends (local file system, GCS)
    containing lower level methods used by the Registry class that are backend-specific.
    """

    @abstractmethod
    def get_registry_proto(self):
        """
        Retrieves the registry proto from the registry path. If there is no file at that path,
        raises a FileNotFoundError.

        Returns:
            Returns either the registry proto stored at the registry path, or an empty registry proto.
        """
        pass

    @abstractmethod
    def update_registry_proto(self, registry_proto: RegistryProto):
        """
        Overwrites the current registry proto with the proto passed in. This method
        writes to the registry path.

        Args:
            registry_proto: the new RegistryProto
        """
        pass

    @abstractmethod
    def teardown(self):
        """
        Tear down all resources.
        """
        pass

File: go/internal/feast/registry/registrystore.go
package registry

import (
	"github.com/feast-dev/feast/go/protos/feast/core"
)

// A RegistryStore is a storage backend for the Feast registry.
type RegistryStore interface {
	GetRegistryProto() (*core.Registry, error)
	UpdateRegistryProto(*core.Registry) error
	Teardown() error
}
